cdap-app-fabric-tests/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -689,17 +702,20 @@ public void testWorkflowScopedArguments() throws Exception {
     oneMRHistoryRuns = getProgramRuns(mr1ProgramId, ProgramRunStatus.COMPLETED);
 
     Id.Program mr2ProgramId = Id.Program.from(TEST_NAMESPACE2, WorkflowAppWithScopedParameters.APP_NAME,
-                                              ProgramType.MAPREDUCE, WorkflowAppWithScopedParameters.ANOTHER_MR);
+                                              applicationDetail.getAppVersion(), ProgramType.MAPREDUCE,
+                                              WorkflowAppWithScopedParameters.ANOTHER_MR);
 
     List<RunRecord> anotherMRHistoryRuns = getProgramRuns(mr2ProgramId, ProgramRunStatus.COMPLETED);
```

Comment:
I think in tests we should use versionless call  "/apps/{app-name}/{program-type}/{program-name}/runs" not versioned "/apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/runs". Generally the versioned call is pretty useless and even harmful. It's important to know which run was for which version, but generally runs should be queried by application, not by application version

Reply:
Also "/apps/{app-name}/{program-type}/{program-name}/runs" has a problem - it retrieves runs only for the latest version. It should return runs for all versions

Reply:
I'll take a look

Reply:
Still applicable

Reply:
so I replaced this with the versionless  `startProgram`